File: java/lib/src/main/java/com/svix/api/BackgroundTaskListOptions.java
// this file is @generated
package com.svix.api;

import com.svix.models.BackgroundTaskStatus;
import com.svix.models.BackgroundTaskType;
import com.svix.models.Ordering;

import lombok.Data;

@Data
public class BackgroundTaskListOptions {
    /** Filter the response based on the status. */
    BackgroundTaskStatus status;

    /** Filter the response based on the type. */
    BackgroundTaskType task;

    /** Limit the number of returned items */
    Long limit;

    /** The iterator returned from a prior invocation */
    String iterator;

    /** The sorting order of the returned items */
    Ordering order;
}

File: kotlin/lib/src/main/kotlin/BackgroundTaskListOptions.kt
package com.svix.kotlin

import com.svix.kotlin.models.BackgroundTaskStatus
import com.svix.kotlin.models.BackgroundTaskType
import com.svix.kotlin.models.Ordering

class BackgroundTaskListOptions : ListOptions() {
    var status: BackgroundTaskStatus? = null
    var task: BackgroundTaskType? = null
    var order: Ordering? = null

    fun order(order: Ordering) = apply { this.order = order }

    fun status(status: BackgroundTaskStatus) = apply { this.status = status }

    fun task(task: BackgroundTaskType) = apply { this.task = task }

    override fun iterator(iterator: String) = apply { super.iterator(iterator) }

    override fun limit(limit: Int) = apply { super.limit(limit) }
}
